src/intra_edge.rs

```diff
@@ -103,58 +160,59 @@ impl EdgeBranch {
     const fn new(edge_flags: EdgeFlags, bl: BlockLevel) -> Self {
         let o = edge_flags;
         let h = [
-            edge_flags | EDGE_LEFT_HAS_BOTTOM,
-            edge_flags & EDGE_LEFT_HAS_BOTTOM,
+            edge_flags.union(EdgeFlags::EDGE_LEFT_HAS_BOTTOM),
+            edge_flags.intersection(EdgeFlags::EDGE_LEFT_HAS_BOTTOM),
         ];
         let v = [
-            edge_flags | EDGE_TOP_HAS_RIGHT,
-            edge_flags & EDGE_TOP_HAS_RIGHT,
+            edge_flags.union(EdgeFlags::EDGE_TOP_HAS_RIGHT),
+            edge_flags.intersection(EdgeFlags::EDGE_TOP_HAS_RIGHT),
         ];
         let node = EdgeNode { o, h, v };
 
         let h4 = [
-            edge_flags | EDGE_LEFT_HAS_BOTTOM,
-            EDGE_LEFT_HAS_BOTTOM
-                | (if bl == BL_16X16 {
-                    edge_flags & EDGE_I420_TOP_HAS_RIGHT
-                } else {
-                    0 as EdgeFlags
-                }),
-            EDGE_LEFT_HAS_BOTTOM,
-            edge_flags & EDGE_LEFT_HAS_BOTTOM,
+            edge_flags.union(EdgeFlags::EDGE_LEFT_HAS_BOTTOM),
+            EdgeFlags::EDGE_LEFT_HAS_BOTTOM.union(if bl == BL_16X16 {
+                edge_flags.intersection(EdgeFlags::EDGE_I420_TOP_HAS_RIGHT)
+            } else {
+                EdgeFlags::empty()
+            }),
+            EdgeFlags::EDGE_LEFT_HAS_BOTTOM,
+            edge_flags.intersection(EdgeFlags::EDGE_LEFT_HAS_BOTTOM),
         ];
 
         let v4 = [
-            edge_flags | EDGE_TOP_HAS_RIGHT,
-            EDGE_TOP_HAS_RIGHT
-                | (if bl == BL_16X16 {
-                    edge_flags & (EDGE_I420_LEFT_HAS_BOTTOM | EDGE_I422_LEFT_HAS_BOTTOM)
-                } else {
-                    0 as EdgeFlags
-                }),
-            EDGE_TOP_HAS_RIGHT,
-            edge_flags & EDGE_TOP_HAS_RIGHT,
+            edge_flags.union(EdgeFlags::EDGE_TOP_HAS_RIGHT),
+            EdgeFlags::EDGE_TOP_HAS_RIGHT.union(if bl == BL_16X16 {
+                edge_flags.intersection(EdgeFlags::union_all([
+                    EdgeFlags::EDGE_I420_LEFT_HAS_BOTTOM,
+                    EdgeFlags::EDGE_I422_LEFT_HAS_BOTTOM,
+                ]))
+            } else {
+                EdgeFlags::empty()
+            }),
```

Comment:
this is the most messy expression. I've tried playing around with it a bit using the laws on union and intersection, but nothing is clearly better to me than this which is at least close to the original.